### PR TITLE
Jetpack: Do not request updates from site if the site is running Jetpack

### PR DESCRIPTION
**Pull request by enejb**
Since Jetpack 4.2 we have better idea as updates sync to us what updates the sites needs. 
This will fix issues related to us hitting jetpack sites all at once. 
Which can cause problems on shared hosts. 

To test:
- Notice the updates link in Site indicator still works as expected. 
- Notice that we do not request updates from Jetpack sites by looking at the network tab in the browser console. 
- Browse around calypso and notice that nothing should be broken.

### CODE REVIEW

File: client/lib/site/index.js [hunk 1]
```diff
@@ -50,11 +50,13 @@ Emitter( Site.prototype );
...
 Site.prototype.attributes = function( attributes ) {
 	attributes = attributes || {};
-
-	for ( var prop in attributes ) {
+	for ( const prop in attributes ) {
 		if ( attributes.hasOwnProperty( prop ) ) {
 			this[ prop ] = attributes[ prop ];
 		}
+		if ( 'updates' === prop ) {
+			this.update = attributes[ prop ];
```
Comment (lezama) [inline client/lib/site/index.js]: I think it would be cleaner to move whenever we use `this.update` to start using `this.updates`.
What do you think?

Comment (enejb) [inline client/lib/site/index.js]: Yes, but do we want to support older version of Jetpack? 
I think this makes it easier to do so. I don't think we use updates anywhere.

**Comment by johnHackworth** [comment]
This look good to me :+1:

**Comment by mtias** [comment]
Do we need to still support the `fetchAvailableUpdates`?

**Comment by johnHackworth** [comment]
looks good to me too! :ship: :rocket:

**Pull request merged**